jkube-kit/generator/webapp/src/main/java/org/eclipse/jkube/generator/webapp/ArtifactUtil.java

```diff
@@ -0,0 +1,71 @@
+package org.eclipse.jkube.generator.webapp;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import org.eclipse.jkube.kit.common.PrefixedLogger;
+
+public class ArtifactUtil {
+  protected static final String CHECKSUM_FILENAME = ".jkube-checksum";
+
+
+  // this is a utility class, so no instances allowed
+  private ArtifactUtil() {
+
+  }
+
+  protected static boolean isArtifactSameAsPrevious(File sourceFile, PrefixedLogger log)
+      throws IOException, NoSuchAlgorithmException {
+    Path path = sourceFile.getParentFile().toPath();
+    byte[] data = Files.readAllBytes(sourceFile.toPath());
+
+    @SuppressWarnings("java:S4790\n") // SHA-1 is not used for security purposes
+    byte[] hash = MessageDigest.getInstance("SHA-1").digest(data);
+
+    String checksum = new BigInteger(1, hash).toString(16);
+    String previousChecksum = null;
+    try {
+      previousChecksum = getPreviousChecksum(path);
+    } catch (ClassNotFoundException e) {
+      log.warn("Failed to read checksum: " + e.getMessage());
+    }
+    log.info("Checksum for %s is %s", sourceFile.getName(), checksum);
+    log.info("Previous checksum for %s is %s", sourceFile.getName(), previousChecksum);
+    boolean isSameChecksum = checksum.equals(previousChecksum);
+    log.info("Same checksum for %s: %s", sourceFile.getName(), isSameChecksum ? "yes" : "no");
+
+    log.info("Saving checksum for %s", path);
+    try {
+      saveChecksum(path, checksum);
+    } catch (IOException e) {
+      log.warn("Failed to save checksum: " + e.getMessage());
+    }
+    return isSameChecksum;
+  }
+
+  protected static void saveChecksum(Path path, String checksum) throws IOException {
+    File f = new File(path.resolve(CHECKSUM_FILENAME).toUri());
+    try (ObjectOutputStream oos = new ObjectOutputStream(Files.newOutputStream(f.toPath()))) {
+      oos.writeObject(checksum);
+      oos.flush();
+    }
+  }
+
+  protected static String getPreviousChecksum(Path path) throws IOException, ClassNotFoundException {
```

Comment:
Is there any reason you've kept visibility as protected? If you want to expose it for testing, default visibility should work as well...